apps/src/redux/unitSelectionRedux.js

```diff
@@ -1,158 +1,69 @@
-import PropTypes from 'prop-types';
-
 // Reducer for script selection in teacher dashboard.
 // Tab specific reducers can import actions from this file
 // if they need to respond to a script changing.
 
-const DEFAULT_SCRIPT_NAME = 'express-2017';
-
 // Action type constants
 export const SET_SCRIPT = 'unitSelection/SET_SCRIPT';
-export const SET_VALID_SCRIPTS = 'unitSelection/SET_VALID_SCRIPTS';
+export const SET_COURSES = 'unitSelection/SET_COURSES';
 
 // Action creators
 export const setScriptId = scriptId => ({type: SET_SCRIPT, scriptId});
-export const setValidScripts = (
-  validScripts,
-  studentScriptIds,
-  validCourses,
-  selectedSection
-) => ({
-  type: SET_VALID_SCRIPTS,
-  validScripts,
-  studentScriptIds,
-  validCourses,
-  selectedSection
+export const setCoursesWithProgress = coursesWithProgress => ({
+  type: SET_COURSES,
+  coursesWithProgress
 });
 
 // Selectors
-export const getSelectedScriptName = state => {
+const getSelectedUnit = state => {
```

Comment:
(unrelated to this change)  I know we're midway through a rename but this file in particular seems to have a mix of script and unit terminology.  Mostly curious, what's our plan to "finish" the rename?  Any sense of what the ends state looks like (e.g. what parts of the system will still use the script terminology?)?

Reply:
> (unrelated to this change) I know we're midway through a rename but this file in particular seems to have a mix of script and unit terminology. Mostly curious, what's our plan to "finish" the rename? 

apologies... this has been forever in my personal backlog. we accidentally started this work at a time when we didn't explicitly make time to finish it. the goal is to finish it this Summer. 

> Any sense of what the ends state looks like (e.g. what parts of the system will still use the script terminology?)?

* in the database, the `scripts` table and various db fields named `script_id` are likely to remain unchanged
* routing will remain on /s/ for now. it's unclear whether controllers will need to remain named scripts_controller in order to keep these routes working
* everything else in dashboard and apps code should change to `unit`

Reply:
I think we are trying to move as much as we can over to use unit. I generally try to move things as I touch them. I stopped short of changing some things like scriptId in this PR because I didn't want the size of it to grow. I think the only things we would not update would be things that are breaking to change such as translation strings. I don't know where we are with the work getting scheduled. @davidsbailey might know more

Reply:
Thanks Dave. I think I was the one that started that work when I was past my due date and looking for things to do while waiting.